GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/MesmerHelper.cs

```diff
@@ -62,61 +65,121 @@ internal static class MesmerHelper
+            .WithBuilds(GW2Builds.May2021Balance, GW2Builds.February2023Balance),
+        new DamageCastFinder(PowerSpike, PowerSpike)
+            .WithBuilds(GW2Builds.February2023Balance),
+        new EXTHealingCastFinder(MantraOfRecovery, MantraOfRecovery)
+            .WithBuilds(GW2Builds.May2021Balance, GW2Builds.February2023Balance),
+        new EffectCastFinderByDst(PowerReturn, EffectGUIDs.MesmerPowerReturn)
+            .UsingDstBaseSpecChecker(Spec.Mesmer)
+            .WithBuilds(GW2Builds.February2023Balance),
+        new EffectCastFinder(MantraOfResolve, EffectGUIDs.MesmerMantraOfResolveAndPowerCleanse)
+            .UsingSrcBaseSpecChecker(Spec.Mesmer)
+            .WithBuilds(GW2Builds.StartOfLife ,GW2Builds.February2023Balance),
+        new EffectCastFinder(PowerCleanse, EffectGUIDs.MesmerMantraOfResolveAndPowerCleanse)
+            .UsingSrcBaseSpecChecker(Spec.Mesmer)
+            .WithBuilds(GW2Builds.February2023Balance, GW2Builds.February2024NewWeapons),
         new EffectCastFinderByDst(PowerCleanse, EffectGUIDs.MesmerMantraOfResolveAndPowerCleanse2)
             .UsingSrcNotBaseSpecChecker(Spec.Mesmer)
             .UsingDstBaseSpecChecker(Spec.Mesmer)
             .UsingSecondaryEffectCheckerInvertedSrc(EffectGUIDs.MesmerMantraOfResolveAndPowerCleanse)
             .WithBuilds(GW2Builds.February2024NewWeapons),
-        new EffectCastFinderByDst(MantraOfConcentration, EffectGUIDs.MesmerMantraOfConcentrationAndPowerBreak).UsingDstBaseSpecChecker(Spec.Mesmer).WithBuilds(GW2Builds.StartOfLife, GW2Builds.February2023Balance),
-        new EffectCastFinderByDst(PowerBreak, EffectGUIDs.MesmerMantraOfConcentrationAndPowerBreak).UsingDstBaseSpecChecker(Spec.Mesmer).WithBuilds(GW2Builds.February2023Balance),
+        new EffectCastFinderByDst(MantraOfConcentration, EffectGUIDs.MesmerMantraOfConcentrationAndPowerBreak)
+            .UsingDstBaseSpecChecker(Spec.Mesmer)
+            .WithBuilds(GW2Builds.StartOfLife, GW2Builds.February2023Balance),
+        new EffectCastFinderByDst(PowerBreak, EffectGUIDs.MesmerMantraOfConcentrationAndPowerBreak)
+            .UsingDstBaseSpecChecker(Spec.Mesmer)
+            .WithBuilds(GW2Builds.February2023Balance),
         // Rifle
         new BuffGiveCastFinder(DimensionalApertureSkill, DimensionalAperturePortalBuff),
         new EffectCastFinder(Abstraction, EffectGUIDs.MesmerRifleAbstraction)
             .UsingSecondaryEffectChecker(EffectGUIDs.MesmerRifleAbstraction2)
             .UsingSrcBaseSpecChecker(Spec.Mesmer),
     ];
-    private static bool SelfHigherHPChecker(DamageEvent x, ParsedEvtcLog log)
+
+    private static bool WithIllusionsChecker(DamageEvent x, ParsedEvtcLog log)
     {
-        double selfHP = x.From.GetCurrentHealthPercent(log, x.Time);
-        double dstHP = x.To.GetCurrentHealthPercent(log, x.Time);
-        if (selfHP < 0.0 || dstHP < 0.0)
+        return x.From == x.CreditedFrom || x.From.IsAnySpecies(Illusions());
```

Comment:
You should do an or on _phanstasms and _lllusions instead of creating a new enumerable at each call.